src/pb-app/src/checkout-form/index.jsx

```diff
@@ -1,41 +1,79 @@
-import React from 'react';
+import React, { useState } from 'react';
 import {ElementsConsumer, CardElement} from '@stripe/react-stripe-js';
 
 import CardSection from '../card-section/index.jsx';
 import ButtonBase from '../button-base/index.jsx';
 import './checkout-form.css';
 
-class CheckoutForm extends React.Component {
+async function paymentSetup(amount) {
+  const response = await fetch('https://api.piggybank.lukejoshuapark.io/payment/setup', {
+    method: 'POST',
+    mode: 'cors', 
+    headers: {
+      'content-type': 'application/json'
+    },
+    redirect: 'follow',
+    referrerPolicy: 'no-referrer',
+    body: JSON.stringify({amount: parseFloat(amount)})
+  });
+  const responseModel =  await response.json();
+  if (response.status !== 200 && !responseModel.error) {
+    responseModel.error = "Internal Server Error :(";
+  }
+  return responseModel;
+}
+
+const CheckoutForm = ({ stripe, elements }) => {
+
+  const [dollarAmount, setDollarAmount] = useState("");
+  const [formMessage, setFormMessage] = useState("");
+  const [clientSecret, setClientSecret] = useState("");
+  const [paymentIntentID, setPaymentIntentID] = useState("");
+
+  const submitClick = async (amount) => {
+    setFormMessage("");
+    if (dollarAmount === "" ) {
+      setFormMessage("Please enter some money!");
+      return;
+    }
+    let responseModel = await paymentSetup(amount)
+    if (responseModel.error) {
+      setFormMessage(responseModel.error);
+      console.log(responseModel.error);
+      return;
+    }
+    console.log("clientSecret = " + responseModel.client_secret + " AND paymentIntentID = " + responseModel.payment_intent_id);
+    setClientSecret(responseModel.client_secret);
+    setPaymentIntentID(responseModel.payment_intent_id);
```

Comment:
Minor again, we probably don't need to store these values in the component state - they'll be used only in the current scope e.g. won't persist past a button click.